test/hotspot/jtreg/runtime/ClassFile/ClassFileVersionTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/**
+ * @test
+ * @bug 8291360
+ * @summary Test getting the class file version for java.lang.Class API
+ * @modules java.base/java.lang:open
+ * @compile classFileVersions.jcod
+ * @run main/othervm --enable-preview ClassFileVersionTest
+ */
+
+import java.lang.reflect.*;
+
+public class ClassFileVersionTest {
+
+    static Method m;
+
+    public static void testIt(String className, int expectedResult) throws Exception {
+        Class<?> testClass = Class.forName(className);
+        int ver = (int)m.invoke(testClass);
+        if (ver != expectedResult) {
+            throw new RuntimeException(
+                "expected " + expectedResult + ", got " + ver + " for class " + className);
+        }
+    }
+
+    public static void main(String argv[]) throws Throwable {
+        Class<?> cl = java.lang.Class.class;
+        m = cl.getDeclaredMethod("getClassFileVersion", new Class[0]);
+        m.setAccessible(true);
+
+        testIt("Version64", 64);
+        testIt("Version59", 59);
```

Comment:
Any particular reason to choose 59? Shouldn't there also be tests for non-zero minor versions?

Reply:
No particular reason for choosing 59 other than it's different and less than 64.  Test case Version64_65535 has a non-zero minor version.

Reply:
I think we should be testing for other real classfile versions that may be encountered, such as 45.3

Reply:
I added a test case for 45.3.